lib/u3d/commands.rb

```diff
@@ -297,14 +298,16 @@ def check_unity_presence(version: nil)
         nil
       end
 
-      def enforce_setup_coherence(packages, options, unity)
+      def enforce_setup_coherence(packages, options, unity, definition)
+        packages = definition.available_packages if options[:all]
         if unity
           UI.important "Unity #{unity.version} is already installed"
-          return false if Helper.linux?
+          # Not needed since Linux custom u3d files contain only one entry wich is Unity
+          # return false if definition.os == :linux
           if packages.include?('Unity')
             UI.important 'Ignoring Unity module, it is already installed'
             packages.delete('Unity')
-            options[:installation_path] ||= unity.path if Helper.windows?
+            options[:installation_path] ||= unity.path if definition.os == :win
```

Comment:
The Unity module is the name of the editor module, so it's required for every version. This line should move to the installer though I think